core/src/test/kotlin/de/fraunhofer/iem/spha/core/strategy/AndKPICalculationStrategyTest.kt

```diff
@@ -0,0 +1,134 @@
+package de.fraunhofer.iem.spha.core.strategy
+
+import de.fraunhofer.iem.spha.core.hierarchy.KpiHierarchyNode
+import de.fraunhofer.iem.spha.model.kpi.KpiId
+import de.fraunhofer.iem.spha.model.kpi.KpiStrategyId
+import de.fraunhofer.iem.spha.model.kpi.RawValueKpi
+import de.fraunhofer.iem.spha.model.kpi.hierarchy.KpiCalculationResult
+import de.fraunhofer.iem.spha.model.kpi.hierarchy.KpiEdge
+import de.fraunhofer.iem.spha.model.kpi.hierarchy.KpiNode
+import kotlin.test.Test
+import org.junit.jupiter.api.Assertions.*
+
+class AndKPICalculationStrategyTest {
+
+    @Test
+    fun emptyEdges() {
+        assertEquals(
+            KpiCalculationResult.Empty(),
+            AndKPICalculationStrategy.calculateKpi(listOf(), strict = true),
+        )
+        assertEquals(
+            KpiCalculationResult.Empty(),
+            AndKPICalculationStrategy.calculateKpi(listOf(), strict = false),
+        )
+    }
+
+    @Test
+    fun calculateCorrectFalse() {
+        val root =
+            KpiHierarchyNode.from(
+                KpiNode(
+                    kpiId = KpiId.ROOT,
+                    kpiStrategyId = KpiStrategyId.XOR_STRATEGY,
+                    edges =
+                        listOf(
+                            KpiEdge(
+                                target =
+                                    KpiNode(
+                                        kpiId = KpiId.NUMBER_OF_SIGNED_COMMITS,
+                                        kpiStrategyId = KpiStrategyId.RAW_VALUE_STRATEGY,
+                                        edges = listOf(),
+                                    ),
+                                weight = 0.5,
+                            ),
+                            KpiEdge(
+                                target =
+                                    KpiNode(
+                                        kpiId = KpiId.NUMBER_OF_COMMITS,
+                                        kpiStrategyId = KpiStrategyId.RAW_VALUE_STRATEGY,
+                                        edges = listOf(),
+                                    ),
+                                weight = 0.5,
+                            ),
+                        ),
+                ),
+                listOf(
+                    RawValueKpi(kind = KpiId.NUMBER_OF_SIGNED_COMMITS, score = 100),
+                    RawValueKpi(kind = KpiId.NUMBER_OF_COMMITS, score = 20),
+                ),
+            )
+
+        val calcRelaxed =
+            AndKPICalculationStrategy.calculateKpi(
+                hierarchyEdges = root.hierarchyEdges,
+                strict = false,
+            )
+
+        val calcStrict =
+            AndKPICalculationStrategy.calculateKpi(
+                hierarchyEdges = root.hierarchyEdges,
+                strict = true,
+            )
+
+        assertEquals(true, calcRelaxed is KpiCalculationResult.Success)
```

Comment:
Shouldn't this be `calcStrict`?
```suggestion
        assertEquals(true, calcStrict is KpiCalculationResult.Success)
```